### ISSUE TITLE
Deprecated react lifecycle hooks in dependencies

### ISSUE THREAD

**Issue by GarvinH** [TBS-EACPD/infobase#739]
The new package-lock.json after adding lodash eslint reveals that react-bootstrap(-typeahead) at minimum need to be updated to use the new lifecycle hooks.

![image](https://user-images.githubusercontent.com/25855114/90283080-855c2000-de3d-11ea-9186-fa56324cca1e.png)

Should wait for #728 to be merged before beginning.

Edit: should also probably wait for #741 to merge as well

**Comment by GarvinH** [comment]
Note: the current react-bootstrap-typeahead has custom implementation done which may not be necessarily addressed in the newer versions.

**Comment by AlexCLeduc** [comment]
I believe the library has fixed this, there should be a [stable version with the unsafe rename](https://github.com/ericgio/react-bootstrap-typeahead/issues/518), and a more recent version that uses [the new lifecycle methods](https://github.com/ericgio/react-bootstrap-typeahead/commit/7d40eee385f04a6bc447e5ff4ca1ac913f9760a7). 

Any reason we can't just bump to version 5? We're on 3 right now

**Comment by Stephen-ONeil** [comment]
It's been locked to 3 for a while because I had to make use of some internal undocumented APIs to get the behaviour we wanted out of the library. Didn't want it rolling forward automatically and breaking in the process. Someone should try rolling forward to 5, checking the release docs for any documented breaking changes. Cross our fingers that those internal APIs didn't change, otherwise it may be hard/impossible to upgrade while keeping the current behaviour.

**Comment by ktw1016** [comment]
This will be a lot more work actually as version 5 drops:
- Bootstrap 3
- React version < 16.8

Not sure about about React upgrade but Bootstrap might be tricky. Related: #84

**Comment by ktw1016** [comment]
This will be a bit more work actually as version 5 drops:
- Bootstrap 3
- React version < 16.8
Not sure about about React upgrade but Bootstrap might be tricky. Related: #84

**Comment by Stephen-ONeil** [comment]
Bootstrap upgrade will be painful. The only thing blocking the React upgrade though, which should be just about trivial otherwise, are the deprecated lifecycles in the typeahead dependency (i.e. this PR, haha).

**Comment by AlexCLeduc** [comment]
I'm confused, what does this issue have to do with lodash-eslint? Why would adding a plugin unrelated to react shed light on a dependency that isn't even covered by our eslint rules?

**Comment by GarvinH** [comment]
The issue is not in lodash-eslint, it's just that for whatever reason, the dev link wouldn't work with the old package-lock.json.
The only way to fix it was to rebuild the package-lock from the package.json.
My assumption was that there was something faulty inside the package-lock???

Anyways, I rebuilt the package-lock.json (the devlink now works) and along with it came these issues that were not showing before.

**Comment by AlexCLeduc** [comment]
I just noticed those warnings in the screenshot are coming from react-devtools, did you add or update that around the same time you took this screenshot? Another possibility is that we didn't pin all of our packages to exact versions, and this upgraded a minor version of a react package, which the devtools are picking up on.


FYI the lock file should be rebuilt each time someone modifies dependencies. If you added a plugin and didn't update the lock-file, then it makes sense for the dev build  to fail when it requests `eslint-plugin-lodash`. CI builds use `npm ci`, short-circuiting `package.json` to use `package-lock.json` directly. If you change dependencies and do an `npm install`, the lock-file should be updated automatically, and you should commit it. 


I don't think these warnings should block us using the plugin, though. We've ignored worse 😄

**Comment by Stephen-ONeil** [comment]
The problem is that the version of the library I pinned us to uses the pre-`UNSAFE__...` version of the deprecated lifecycle methods. Those will stop working in React 17. We might be able to just roll the typeahead forward, I only locked it because I made use of undocumented internal APIs, and didn't want to chance them changing out from under me during a minor update.

... the other problem is updating the typeahead to the latest version requires a bootstrap upgrade. Possible there's some mid-version of the component that doesn't use the deprecated lifecycle names but still uses Bootstrap 4, haven't checked. Otherwise it's a question of replacing the typeahead or finally getting on with updating bootstrap. Something's gotta move though, to unblock this dependency traffic-jam.

**Comment by GarvinH** [comment]
Interestingly, adding the "essentials" of eslint lodash didn't cause any issues with the package-lock.json
I wonder if npm downloaded a corrupt/bad file or something...

**Comment by GarvinH** [comment]
Bootstrap finally has been moved to 4!
And a react upgrade came with it.

With that, Nivo is now out of date (and it doesn't seem to be an easy update since the props have supposedly changed without much docs?
This might be an opportunity to move to a more stable(?) react data visualizer such as `react-vis` for example?

React-aria-modal will also require an update to solve the deprecation warnings (when the update is finally completed on their end).

**Comment by Stephen-ONeil** [comment]
@GarvinH Close this issue please. Open a new one (high priority) for Nivo, and another as a reminder to keep an eye on `react-aria-modal`.

Updating Nivo needs to be high priority because the big wall of react warnings it spits out right now are going to be obscuring other more important warnings, and after a while we'll all get notification-blindness to any React dev tools warnings.

**Comment by Stephen-ONeil** [comment]
Closing in favour of more specific issues #909 and #910
